src/Avalonia.Base/Media/Geometry.cs

```diff
@@ -117,9 +118,10 @@ public bool StrokeContains(IPen pen, Point point)
         /// </remarks>
         protected static void AffectsGeometry(params AvaloniaProperty[] properties)
         {
+            var invalidateObserver = new AnonymousObserver<AvaloniaPropertyChangedEventArgs>(AffectsGeometryInvalidate);
             foreach (var property in properties)
             {
-                property.Changed.Subscribe(AffectsGeometryInvalidate);
+                property.Changed.Subscribe(invalidateObserver);
```

Comment:
Why can't we keep an extension method that accepts a delegate? It was kinda convenient to use

Reply:
It's still there. I just don't like that we create an observer on each loop iteration for each property instead of reusing single one.